// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.helloworld";
option java_outer_classname = "HelloWorldProto";
option objc_class_prefix = "HLW";

package helloworld;

message SliceRequest {
  uint64 slice_partition = 1;
  uint64 data_start = 2;
  uint64 data_len = 3;
}

message Request {
  repeated SliceRequest slice_request = 1;
}

message Response {
  int32 status = 1; // 0，返回结果ok，-1 服务发生内部错误
  repeated bytes slice_data = 2;
}

service Greeter {
  rpc Get(Request) returns (Response) {}
}